doc/Programming-guide/Your-first-Kaa-application/index.md

```diff
@@ -2,7 +2,615 @@
+    (*configuration-schema.json*) scheme:
+
+    ![Data collection scheme](attach/new_config_schema.png)
+
+As a result of these operations you will see configuration and data collections
+schemes in the list:
+![Data collection schema complete](attach/log_schema_list.png)
+![Configuraion schema complete](attach/config_schema_list.png)
+
+In this screenshot, version 2.0 is log and configuraion schema version that
+was just created. We will use this version for the SDK generation later.
+
+# Log appenders setup
+
+In order to use Data Collection feature it is required to setup **Log Appender**.
+In this example we will use MongoDB log appender.
+Refer to [Admin UI section describing how to setup it]({{root_url}}/Administration-guide/Tenants-and-applications-management/#mongodb-log-appender).
+
+# Generating SDK
+
+To generate an SDK for the new application, proceed as follows:
+
+1. Select the **My First Kaa Application** application and click **Generate SDK**.
+
+    ![SDK generation](attach/generate_sdk.png)
+1. Click **Add SDK profile**.
+
+    ![Add SDK](attach/add_sdk.png)
+1. In the **Add SDK profile** window, fill in the fields as shown in the following
+screenshot  and then click **Add**.
+
+    ![Configure SDK](attach/configure_sdk.png)
+1. Now the SDK is configured and ready to be generated. Click **Generate SDK** for
+corresponding SDK profile. In the **Generate SDK** window select the target platform for
+your SDK and click Generate SDK.
+
+    ![Generate SDK](attach/generate_configured_sdk.png)
+
+    ![Generate SDK](attach/select_target_platform.png)
+
+After the SDK is generated, you will be presented with a window asking you to save a
+.jar file with the generated SDK (for Java) or an archive with the generated SDK (for C,
+C++ or Objective-C). Specify the file name and location on your computer and then click
+**Save**. The SDK is now downloaded to your computer.
+
+Note that in this example we are generating the SDK based on the default profile and
+notification schemas. These schemas are automatically populated during the creation of
+the application.
+If necessary, you can overwrite them using [Admin UI]({{root_url}}/Administration-guide/Tenants-and-applications-management/).
+
+# Sample client application
+
+Once you have downloaded the SDK, you can use it in your sample project. The following
+code block illustrates a simple desktop application that will send virtual temperature
+data from the Kaa endpoint with required configuration.
+
+NOTE: After generating the C++/Objective-C SDKs, you need to build them before creating the application.
+
+## Preparation
+
+<ul class="nav nav-tabs">
+  <li class="active"><a data-toggle="tab" href="#prep-c">C SDK</a></li>
+  <li><a data-toggle="tab" href="#prep-java">Java SDK</a></li>
+</ul>
+
+<div class="tab-content">
+
+<div id="prep-c" class="tab-pane fade in active" markdown="1" >
+
+Before you start with C application code, CMake file must be filled.
```

Comment:
Please also add information on the required packages like cmake and libssl-dev (or give a link to the appropriate documentation page).